lib/querystring.js

```diff
@@ -18,7 +18,7 @@ QueryString.unescapeBuffer = function(s, decodeSpaces) {
   var n, m, hexchar;
 
   for (var inIndex = 0, outIndex = 0; inIndex <= s.length; inIndex++) {
-    var c = s.charCodeAt(inIndex);
+    var c = inIndex < s.length ? s.charCodeAt(inIndex) : NaN;
```

Comment:
`charCodeAt()` returns `NaN` for out of bounds indices. I was just keeping the same behavior here but avoiding the deopt.